ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryPlan.java

```diff
@@ -33,12 +33,14 @@
   private final Optional<SourceName> sink;
   private final ExecutionStep<?> physicalPlan;
   private final QueryId queryId;
+  private final Optional<String> usesSharedRuntime;
 
   public QueryPlan(
       @JsonProperty(value = "sources", required = true) final Set<SourceName> sources,
       @JsonProperty(value = "sink") final Optional<SourceName> sink,
       @JsonProperty(value = "physicalPlan", required = true) final ExecutionStep<?> physicalPlan,
-      @JsonProperty(value = "queryId", required = true) final QueryId queryId
+      @JsonProperty(value = "queryId", required = true) final QueryId queryId,
+      @JsonProperty(value = "sharedRuntimes") final Optional<String> usesSharedRuntime
```

Comment:
we should probably name this something different right? is the intent to store the runtime ID in there? maybe call it runtimeId then?